### PR TITLE
Make recommended fee confirmation target configurable

### PR DESCRIPTION
**Pull request by bolatovumar**
Address #1110

### CODE REVIEW

**Comment by bolatovumar** [comment]
Thanks to @rockstardev for going over the codebase with me and helping with this.

File: BTCPayServer/Models/StoreViewModels/CheckoutExperienceViewModel.cs [hunk 1]
```diff
@@ -40,6 +40,10 @@ public class Format
         [Display(Name = "Show recommended fee")]
         public bool ShowRecommendedFee { get; set; }
 
+        [Display(Name = "Recommended fee confirmation target blocks")]
```
Comment (NicolasDorier) [inline BTCPayServer/Models/StoreViewModels/CheckoutExperienceViewModel.cs]: add `(in satoshi per byte)`

**Comment by NicolasDorier** [comment]
Cool. The only bad point is that this also impact the network fee which will become way more expensive.

**Comment by bolatovumar** [comment]
@rockstardev this is what i was trying to get at in our session but couldn't quite express as I wasn't sure about how the whole network fee thing works. I wanted to have the network fee still target 20 blocks (leave it hardcoded) but have the invoice fee be configurable. Maybe something we take a look at next time?

**Comment by rockstardev** [comment]
This won't impact network fee because when those other controllers call GetFeeRateAsync(), they don't specify parameter and for them default is 20.

https://github.com/btcpayserver/btcpayserver/pull/1127/commits/0a9290a980f87096baaac2957c19611f72eccb52#diff-8a2c0e8027c709ab0fb4ed0996572b0cR39

Analyzing this further, I think we made mistake with introducing new setting - target fee rate for invoices should be whatever number of confirmations merchants has already set in their store.

I wish I have time to refactor this... will add it to the list of stuff to work on when I have time.

**Comment by bolatovumar** [comment]
@rockstardev not sure what you mean by "whatever number of confirmations merchants has already set in their store"? There is currently no setting for how many blocks a merchant wants to have the recommended fee target. There is a setting for how many blocks a merchants wants to wait for before the invoice is considered paid but they may wanna set that value to be something different. These are not the same things the way I see it.

@NicolasDorier does rockstar's reasoning make sense to you?

**Comment by NicolasDorier** [comment]
fixed with https://github.com/btcpayserver/btcpayserver/commit/163ef031b97e43a449c4b78ba97c211aecfcb91b

**Comment by NicolasDorier** [comment]
and https://github.com/btcpayserver/btcpayserver/commit/9ff9377bc7314f2b7395e37a49b8c4b34f2803ae

**Comment by bolatovumar** [comment]
Thanks @NicolasDorier, looks like we are good now.

**Pull request merged**